AWSPinpointTests/AWSPinpointEventRecorderTests.m

```diff
@@ -305,6 +305,83 @@ - (void) testUpdateSessionCampaignAttributes {
+    [config.userDefaults removeObjectForKey:AWSPinpointSessionKey];
+    [config.userDefaults synchronize];
+    
+    
+    AWSPinpoint *pinpoint = [AWSPinpoint pinpointWithConfiguration:config];
+    [pinpoint.sessionClient stopSession];
+    [pinpoint.analyticsClient removeAllGlobalFeatureAttributes];
+    [[pinpoint.analyticsClient.eventRecorder removeAllEvents] waitUntilFinished];
+    
+    NSData *data = [pinpoint.pinpointContext.configuration.userDefaults dataForKey:AWSPinpointSessionKey];
+    AWSPinpointSession *session = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+    
+    AWSDDLogError(@"Session Object Should be Empty: %@",session.description);
+    
+    XCTAssertNil([pinpoint.pinpointContext.configuration.userDefaults dataForKey:AWSPinpointSessionKey]);
+    
+    [[pinpoint.sessionClient startSession] continueWithBlock:^id _Nullable(AWSTask * _Nonnull task) {
+        [[pinpoint.analyticsClient.eventRecorder getCurrentSession:pinpoint.sessionClient.session] continueWithBlock:^id _Nullable(AWSTask * _Nonnull task) {
+            XCTAssertNotNil(task.result);
+            XCTAssertTrue([task.result isKindOfClass:[AWSPinpointEvent class]]);
+            AWSPinpointEvent *sessionStartResult = task.result;
+            XCTAssertTrue([sessionStartResult.eventType isEqualToString:@"_session.start"]);
+            XCTAssertTrue([sessionStartResult.session.sessionId isEqualToString:pinpoint.sessionClient.session.sessionId]);
+            XCTAssertTrue([sessionStartResult.session.startTime.description isEqualToString:pinpoint.sessionClient.session.startTime.description]);
+            XCTAssertEqual(sessionStartResult.allAttributes.count, 0);
+            
+            [[pinpoint.analyticsClient.eventRecorder updateSessionStartWithFeatureAttributes:@{@"campaignAttrKey":@"campaignAttrVal"}] continueWithBlock:^id _Nullable(AWSTask * _Nonnull task) {
+                [[pinpoint.analyticsClient.eventRecorder getCurrentSession:pinpoint.sessionClient.session] continueWithBlock:^id _Nullable(AWSTask * _Nonnull task) {
+                    XCTAssertNotNil(task.result);
+                    XCTAssertTrue([task.result isKindOfClass:[AWSPinpointEvent class]]);
+                    AWSPinpointEvent *sessionStartResult = task.result;
+                    XCTAssertTrue([sessionStartResult.eventType isEqualToString:@"_session.start"]);
+                    XCTAssertTrue([sessionStartResult.session.sessionId isEqualToString:pinpoint.sessionClient.session.sessionId]);
+                    XCTAssertTrue([sessionStartResult.session.startTime.description isEqualToString:pinpoint.sessionClient.session.startTime.description]);
+                    XCTAssertEqual(sessionStartResult.allAttributes.count, 1);
+                    XCTAssertTrue([sessionStartResult.allAttributes[@"campaignAttrKey"] isEqualToString:@"campaignAttrVal"]);
+                    
+                    // Overwrite  with journey attributes
```

Comment:
remove double space between `Overwrite` and `with`

Reply:
Will Remove